app/Controllers/Discussions/TagController.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Controllers\Discussions;
+
+use App\Controllers\BaseController;
+use App\Models\ThreadModel;
+use InvalidArgumentException;
+
+/**
+ * Class Home
+ */
+class TagController extends BaseController
+{
+    /**
+     * Display a standard forum-style list of discussions.
+     */
+    public function get(string $tagSlug)
+    {
+        $table = [
```

Comment:
At some point, I'd love to refactor all of our discussion searching into a `search` method on the model to clean it up a little, but that's just me thinking out loud.

Reply:
On the model level, we're already using only one method: `ThreadModel::forList`.

Do you mean searching for threads, in the top menu bar? Or one method in the controller to display threads list, tags, and categories?

Reply:
I just meant handling all of the filtering, sorting, etc. It feels like we're using similar code in several places, but maybe they're all different models and wasn't thinking straight last night :)

Reply:
I made some refactoring today, and I see redundant code in the DiscussionController... however, for now, I think I will leave it as it is until we are sure the functionality is finished.